torchao/prototype/low_bit_optim/cpu_offload.py

```diff
@@ -38,51 +40,57 @@ def __init__(
         if not isinstance(param_groups[0], dict):
             param_groups = [{"params": param_groups}]
 
-        self.param_cuda2cpu_map = dict()
+        self.param_d2h_map = dict()
         self.optim_dict = dict()
-        self.stream = torch.cuda.Stream()
+        self.device = device
+        self.stream = torch.Stream(device=self.device)
```

Comment:
2.3 CI is failing because `torch.Stream` is only available on 2.5 I think. Perhaps we can do sth like `getattr(torch, device_type).Stream()`? (maybe do the same thing for other similar device API calls too?) Or just make this an if-else statement.

Reply:
Replace related if-else statement w/ `getattr(torch, device_type).foo()`. And checked PyTorch v2.3 do have `torch.Stream` API and works well. Not sure the reason of 2.3 CI failure from the log. Since there are several updates about device setting, maybe we can trigger the CI again? THX for ur review comment.